packages/build/tests/core/tests.js

```diff
@@ -320,3 +322,30 @@ if (!version.startsWith('v8.')) {
     await runFixture(t, 'functions_user_missing')
   })
 }
+
+test('Does not generate a `manifest.json` file when the feature flag is not enabled', async (t) => {
+  const fixtureName = 'functions_internal_no_manifest_2'
+
+  await removeDir(`${FIXTURES_DIR}/${fixtureName}/.netlify/functions`)
+  await runFixture(t, fixtureName, { flags: { mode: 'buildbot' } })
+
+  t.false(await pathExists(`${FIXTURES_DIR}/${fixtureName}/.netlify/functions/manifest.json`))
+})
+
+test('Does not generate a `manifest.json` file when running in buildbot', async (t) => {
+  const fixtureName = 'functions_internal_no_manifest_1'
+
+  await removeDir(`${FIXTURES_DIR}/${fixtureName}/.netlify/functions`)
+  await runFixture(t, fixtureName, { flags: { featureFlags: { functionsBundlingManifest: true }, mode: 'buildbot' } })
+
+  t.false(await pathExists(`${FIXTURES_DIR}/${fixtureName}/.netlify/functions/manifest.json`))
+})
+
+test('Generates a `manifest.json` file when running outside of buildbot', async (t) => {
+  const fixtureName = 'functions_internal_manifest'
+
+  await removeDir(`${FIXTURES_DIR}/${fixtureName}/.netlify/functions`)
+  await runFixture(t, fixtureName, { flags: { featureFlags: { functionsBundlingManifest: true }, mode: 'cli' } })
+
+  t.true(await pathExists(`${FIXTURES_DIR}/${fixtureName}/.netlify/functions/manifest.json`))
```

Comment:
What are your thoughts on also ensuring this file can be read and JSON-parsed? ZISI already tests this, but this would provide with an additional layer of integration test.

Reply:
Good idea! Done in https://github.com/netlify/build/pull/3378/commits/d17193d724c331220c409949b8986166db0fa3ba.